.github/workflows/pr-release-warning.yml

```diff
@@ -0,0 +1,25 @@
+name: Warn Release Pull Requests
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+jobs:
+  Check:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps: 
+      - name: check branch name
+        id: check-release
+        run: |
+          if [[ ${{ github.head_ref }} =~ ^release/([0-9]+\.[0-9]+\.[0-9]+$) ]]; then
```

Comment:
I think you can push the event with the branch `release/`, for instance, see:

https://github.com/elastic/apm-server/blob/d9a0bbfe398da574845cee95bc069e4bf671ba9f/.github/workflows/update-dependabot-pr.yml#L7-L9

This should avoid situation of running for all the PRs regardless

Reply:
That's understandable, however, it looks like the `branches` option on `pull_request` trigger only applies to the target branch, not the head branch. https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-branches

If you believe this is too expensive of a job I can just remove it. I only provides a bit of sanity check to PRs to warn if it's going to trigger additional jobs.